src/status_im/ui/screens/mobile_network_settings/events.cljs

```diff
@@ -0,0 +1,101 @@
+(ns status-im.ui.screens.mobile-network-settings.events
+  (:require [status-im.utils.handlers :as handlers]
+            [status-im.accounts.update.core :as accounts.update]
+            [status-im.utils.fx :as fx]
+            [status-im.ui.components.bottom-sheet.core :as bottom-sheet]
+            [status-im.accounts.db :as accounts.db]
+            [status-im.ui.screens.navigation :as navigation]
+            [status-im.mailserver.core :as mailserver]))
+
+(defn cellular? [network-type]
+  (= network-type "cellular"))
+
+(fx/defn sheet-defaults
+  [{:keys [db]}]
+  (let [remember-choice? (get-in db [:account/account :remember-syncing-choice?])]
+    {:db (assoc db :mobile-network/remember-choice? remember-choice?)}))
+
+(fx/defn on-network-status-change
+  [{:keys [db] :as cofx}]
+  (let [logged-in? (accounts.db/logged-in? cofx)
+        {:keys [remember-syncing-choice?]} (:account/account db)]
+    (apply
+     fx/merge
+     cofx
+     (cond
+       (and logged-in?
+            (cellular? (:network/type db))
+            (not remember-syncing-choice?)
+            (not= :create-account (:view-id db)))
+
+       [(bottom-sheet/show-bottom-sheet
+         {:view :mobile-network})
+        (sheet-defaults)]
+
+       logged-in?
+       [(bottom-sheet/hide-bottom-sheet)]))))
+
+(defn apply-settings
```

Comment:
Probably it would be good to add some tests for this namespace, also I generally prefer not to use `trooleans` (nil, true, false), at which point I'd rather use strings to capture state

Reply:
do you mean for `syncing-on-mobile-network?` and `remember-syncing-choice?`? These fields are not nillable

Reply:
`remember?` and there's another example I think in a different file `new-show?` where we explicitly check that is `false?`, although in that  case looks we have less control as coming from props

Reply:
re `remember?`, is it better now or you want it to be something like `:default`, `:true`, `:false`?

Reply:
I'd rather have it as the same type (`:not-answered :true :false`), rather than :default true false

Reply:
well, it means that in [some places](https://github.com/status-im/status-react/blob/54d26a723d989341b0d84ce65d7feffbd7f318c3/src/status_im/ui/screens/mobile_network_settings/events.cljs#L75 ) we need to convert boolean to this :true/:false thing, I realize the concern around troolean thing, but I have doubts it is worth doing here

Reply:
oh I see, never mind then, probably not worth